mm2src/mm2_main/src/lp_wallet.rs

```diff
@@ -550,3 +532,64 @@ pub async fn get_wallet_names_rpc(ctx: MmArc, _req: Json) -> MmResult<GetWalletN
         activated_wallet: activated_wallet.clone(),
     })
 }
+
+/// `SeedPasswordUpdateRequest` represents a request to update
+/// the password for the seed storage.
+/// It includes the current password and the new password to be set.
+#[derive(Debug, Deserialize)]
+pub struct SeedPasswordUpdateRequest {
+    /// The current password for the seed storage.
+    pub current_password: String,
+    /// The new password to replace the current password.
+    pub new_password: String,
+}
+
+/// `SeedPasswordUpdateResponse` represents the result of a
+/// password update request.
+/// It contains a boolean indicating whether the operation was successful.
+#[derive(Serialize)]
+pub struct SeedPasswordUpdateResponse {
+    /// `true` if the password update was successful, `false` otherwise.
+    successful: bool,
+}
+
+/// RPC function to handle a request for updating the seed storage password.
+///
+/// # Arguments
+/// - `ctx`: The shared context (`MmArc`) for the application.
+/// - `req`: The `SeedPasswordUpdateRequest` containing the current and new passwords.
+///
+/// # Example
+/// ```ignore
+/// let request = SeedPasswordUpdateRequest {
+///     current_password: "old_password".to_string(),
+///     new_password: "new_password".to_string(),
+/// };
+/// let response = update_seed_storage_password_rpc(ctx, request).await?;
+/// assert!(response.successful);
+/// ```
+pub async fn update_seed_storage_password_rpc(
+    ctx: MmArc,
+    req: SeedPasswordUpdateRequest,
+) -> MmResult<SeedPasswordUpdateResponse, WalletsStorageRpcError> {
+    let wallet_name = ctx
+        .wallet_name
+        .get()
+        .ok_or(WalletsStorageRpcError::Internal(
+            "`wallet_name` not initialized yet!".to_string(),
+        ))?
+        .clone()
+        .ok_or_else(|| WalletsStorageRpcError::Internal("`wallet_name` cannot be None!".to_string()))?;
+    // read mnemonic for a wallet_name using current user's password.
+    let mnemonic = read_and_decrypt_passphrase_if_available(&ctx, &req.current_password)
+        .await?
+        .ok_or(MmError::new(WalletsStorageRpcError::Internal(format!(
+            "{wallet_name}: wallet mnemonic file not found"
+        ))))?;
+    // encrypt mnemonic with new passphrase.
+    let encrypted_data = encrypt_mnemonic(&mnemonic, &req.new_password)?;
+    // save new encrypted mnemonic data with new password
+    save_encrypted_passphrase(&ctx, &wallet_name, &encrypted_data).await?;
+
```

Comment:
could u actually verify whether this would play well in wasm?
this call eventually does [`table.add_item`](https://github.com/KomodoPlatform/komodo-defi-framework/blob/ac926dda0a89b239f753eae833ced97a3b703544/mm2src/mm2_main/src/lp_wallet/mnemonics_wasm_db.rs#L115), which if the item already exists will error `ConstraintError` (as per [this](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/add)).

We should use `table.replace_item` to be able to supported updates.

Reply:
done thank you